app/src/main/java/org/jellyfin/mobile/webapp/ConnectionHelper.kt

```diff
@@ -0,0 +1,207 @@
+                    connect()
+                    true
+                }
+                else -> false
+            }
+        }
+        connectButton.setOnClickListener {
+            connect()
+        }
+
+        // Show keyboard
+        serverSetupLayout.doOnNextLayout {
+            hostInput.postDelayed(25) {
+                hostInput.requestFocus()
+                activity.getSystemService<InputMethodManager>()?.showSoftInput(hostInput, InputMethodManager.SHOW_IMPLICIT)
+            }
+        }
+    }
+
+    private fun connect() {
+        hostInput.isEnabled = false
+        connectButton.isEnabled = false
+        clearConnectionError()
+        activity.lifecycleScope.launch {
+            val httpUrl = checkServerUrlAndConnection(hostInput.text.toString())
+            if (httpUrl != null) {
+                appPreferences.instanceUrl = httpUrl.toString()
+                cachedInstanceUrl = httpUrl
+                rootView.removeView(serverSetupLayout)
+                loadOrShowSetup()
+            }
+            hostInput.isEnabled = true
+            connectButton.isEnabled = true
+        }
+    }
+
+    private fun showConnectionError(@StringRes errorString: Int = R.string.connection_error_cannot_connect) {
+        connectionErrorText.setText(errorString)
+        connectionErrorText.isVisible = true
+    }
+
+    private fun clearConnectionError() {
+        connectionErrorText.isVisible = false
+    }
+
+    private suspend fun checkServerUrlAndConnection(enteredUrl: String): HttpUrl? {
+        val normalizedUrl = enteredUrl.run {
+            if (lastOrNull() == '/') this
+            else "$this/"
+        }
+
+        val urls = when {
+            normalizedUrl.startsWith("http") -> listOf(normalizedUrl)
+            else -> listOf("https://$normalizedUrl", "http://$normalizedUrl")
+        }
+
+        var httpUrl: HttpUrl? = null
+        var serverInfoResponse: String? = null
+        loop@ for (url in urls) {
+            httpUrl = url.toHttpUrlOrNull()
+
+            if (httpUrl == null) {
+                showConnectionError(R.string.connection_error_invalid_format)
+                return null // Format is invalid, don't try any other variants
+            }
+
+            serverInfoResponse = fetchServerInfo(activity.httpClient, httpUrl)
+            if (serverInfoResponse != null)
+                break@loop
+        }
+
+        if (httpUrl == null || serverInfoResponse == null) {
+            showConnectionError()
+            return null
+        }
+
+        val isValidInstance = try {
+            val serverInfo = JSONObject(serverInfoResponse)
+            val version = serverInfo.getString("Version")
+                .split('.')
+                .mapNotNull(String::toIntOrNull)
+            when {
+                version.size != 3 -> false
+                version[0] == 10 && version[1] < 3 -> true // Valid old version
```

Comment:
We might want to add the "supported version" as a constant somewhere so it can be changed more easily.

Reply:
I think the check is only there because of the "ProductName" variable that isn't available before 10.3..

Reply:
Hmm, I can imagine that some features in the nativeshell (and in the future, the apiclient) need a minimum version set too. Might even be a good thing to require 10.6.x at the release too.